awx/main/models/schedules.py

```diff
@@ -207,23 +246,33 @@ def rrulestr(cls, rrule, fast_forward=True, **kwargs):
                 raise ValueError('A valid TZID must be provided (e.g., America/New_York)')
 
         # Fast forward is a way for us to limit the number of events in the rruleset
-        # If we are fastforwading and we don't have a count limited rule that is minutely or hourley
+        # If we are fast forwarding and we don't have a count limited rule that is minutely or hourly
         # We will modify the start date of the rule to last week to prevent a large number of entries
         if fast_forward:
             try:
                 # All rules in a ruleset will have the same dtstart value
-                #   so lets compare the first event to now to see if its > 7 days old
+                # so lets compare the first event to now to see if its > 30 days old
+                # we choose > 30 days because if rrule has FREQ=HOURLY and INTERVAL=23, it will take
+                # at least 23 days before we can fast forward reliably and retain stable occurrences.
+                # since we are fast forwarding to 7 days ago, we check fo rrrules older than (23+7) days
                 first_event = x[0]
-                if (now() - first_event).days > 7:
+                if (now() - first_event).days > 30:
                     for rule in x._rrule:
                         # If any rule has a minutely or hourly rule without a count...
                         if rule._freq in [dateutil.rrule.MINUTELY, dateutil.rrule.HOURLY] and not rule._count:
                             # hourly/minutely rrules with far-past DTSTART values
                             # are *really* slow to precompute
                             # start *from* one week ago to speed things up drastically
-                            new_start = (now() - datetime.timedelta(days=7)).strftime('%Y%m%d')
-                            # Now we want to repalce the DTSTART:<value>T with the new date (which includes the T)
-                            new_rrule = re.sub('(DTSTART[^:]*):[^T]+T', r'\1:{0}T'.format(new_start), rrule)
+                            try:
+                                new_start = fast_forward_date(rule)
+                            except RuntimeError as e:
```

Comment:
Since there's nothing wrong with the runtime, this would also need to align with the above suggestions to represent the case with a different exception:
```suggestion
                            except ValueError as val_err:
```

Reply:
no longer throwing error